freqtrade/tests/test_hyperopt.py

```diff
@@ -0,0 +1,166 @@
+        if params['mfi']['enabled']:
+            conditions.append(dataframe['mfi'] < params['mfi']['value'])
+        if params['fastd']['enabled']:
+            conditions.append(dataframe['fastd'] < params['fastd']['value'])
+        if params['adx']['enabled']:
+            conditions.append(dataframe['adx'] > params['adx']['value'])
+        if params['cci']['enabled']:
+            conditions.append(dataframe['cci'] < params['cci']['value'])
+        if params['over_sar']['enabled']:
+            conditions.append(dataframe['close'] > dataframe['sar'])
+        if params['uptrend_sma']['enabled']:
+            prevsma = dataframe['sma'].shift(1)
+            conditions.append(dataframe['sma'] > prevsma)
+
+        prev_fastd = dataframe['fastd'].shift(1)
+        # TRIGGERS
+        triggers = {
+            'lower_bb': dataframe['tema'] <= dataframe['blower'],
+            'faststoch10': (dataframe['fastd'] >= 10) & (prev_fastd < 10),
+        }
+        conditions.append(triggers.get(params['trigger']['type']))
+
+        dataframe.loc[
+            reduce(lambda x, y: x & y, conditions),
+            'buy'] = 1
+        dataframe.loc[dataframe['buy'] == 1, 'buy_price'] = dataframe['close']
+
+        return dataframe
+    return populate_buy_trend
+
+@pytest.mark.skipif(not os.environ.get('BACKTEST', False), reason="BACKTEST not set")
+def test_hyperopt(conf, pairs, mocker):
+
+    def optimizer(params):
+        return backtest(conf, pairs, mocker, buy_strategy_generator(params))
+
+    space = {
+        'mfi': hp.choice('mfi', [
+            {'enabled': False},
+            {'enabled': True, 'value': hp.uniform('mfi-value', 2, 40)}
+        ]),
+        'fastd': hp.choice('fastd', [
+            {'enabled': False},
+            {'enabled': True, 'value': hp.uniform('fastd-value', 2, 40)}
+        ]),
+        'adx': hp.choice('adx', [
+            {'enabled': False},
+            {'enabled': True, 'value': hp.uniform('adx-value', 2, 40)}
+        ]),
+        'cci': hp.choice('cci', [
+            {'enabled': False},
+            {'enabled': True, 'value': hp.uniform('cci-value', -200, -100)}
+        ]),
+        'below_sma': hp.choice('below_sma', [
+            {'enabled': False},
+            {'enabled': True}
+        ]),
+        'over_sma': hp.choice('over_sma', [
+            {'enabled': False},
+            {'enabled': True}
+        ]),
+        'over_sar': hp.choice('over_sar', [
+            {'enabled': False},
+            {'enabled': True}
+        ]),
+        'uptrend_sma': hp.choice('uptrend_sma', [
+            {'enabled': False},
+            {'enabled': True}
+        ]),
+        'trigger': hp.choice('trigger', [
+            {'type': 'lower_bb'},
+            {'type': 'faststoch10'}
+        ]),
+    }
+
+    # print(hyperopt.pyll.stochastic.sample(space))
+    print('Best parameters {}'.format(fmin(fn=optimizer, space=space, algo=tpe.suggest, max_evals=40)))
```

Comment:
Should the `max_evals` be a "bit" smaller by default. In my machine only `10` evals takes ~190s.

Reply:
Well the test is also disabled by default. You need `BACKTEST=true` to run it.